### ISSUE TITLE
[no-restricted-imports] No way to forbid only regular imports from a package & allow type imports

### ISSUE THREAD

**Issue by mgol** [typescript-eslint/typescript-eslint#2403]
- [x] I have first restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "sourceType": "module"
  },
  "plugins": ["@typescript-eslint/eslint-plugin"],
  "rules": {
    "no-restricted-imports": ["error", {
        "patterns": ["/forbidden-package"]
    }]
  }
}
```

```TS
import something from 'forbidden-package';
```

**Expected Result**

I expected to be able to apply different rules to type imports.

**Actual Result**

There's no way of allowing only type imports from a specific package. An ESLint core rule can't differentiate between them as pure JS doesn't have the `import type` concept.

**Additional Info**

The issue is that a core ESLint rule cannot distinguish between type imports & regular imports since there is no type import concept in pure JavaScript. To fix this, we need a new `typescript-eslint` rule that doesn't exist yet (from what I could see).

If such a rule existed, I'd also expect it to be mentioned in https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md as the replacement for TSLint's `import-blacklist` instead of the current core ESLint `no-restricted-imports` rule.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.9.1` |
| `@typescript-eslint/parser`        | `3.9.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `14.8.0` |

**Comment by nicu-chiciuc** [comment]
A good use case for this rule is Meteor. Most of the code is isomorphic but if a file is somewhere inside a `/server/` directory it can only be imported from a another file in a `/server/` directory. Same applies to `/client/` directories.

**Comment by stefnotch** [comment]
For those coming from Google, the relevant upstream issue used to be https://github.com/eslint/eslint/issues/13758

**Comment by stefnotch** [comment]
And there is also a workaround rule:

If you want to disallow imports from `../dist`, then the rule would be
```js
"no-restricted-syntax": [
      "error",
      {
        selector:
          "ImportDeclaration[importKind!='type'][source.value=/..\\u002Fdist/]",
        message: "Must use 'import type' when importing from '../electron'.",
      },
    ],
```
(The `\\u002F` has to be used instead of the forwards slash because of https://github.com/estools/esquery/issues/68 )

**Comment by Josh-Cena** [comment]
Should be fixed by #3775
